lib/web_ui/dev/firefox_installer.dart

```diff
@@ -10,6 +10,59 @@ import 'package:path/path.dart' as path;
 import 'common.dart';
 import 'environment.dart';
 
+/// Returns the installation of Firefox, installing it if necessary.
+///
+/// If [requestedVersion] is null, uses the version specified on the
+/// command-line. If not specified on the command-line, uses the version
+/// specified in the "browser_lock.yaml" file.
+///
+/// If [requestedVersion] is not null, installs that version. The value
+/// may be "latest" (the latest stable Firefox version), "system"
+/// (manually installed Firefox on the current operating system), or an
+/// exact version number such as 69.0.3. Versions of Firefox can be found here:
+///
+/// https://download-installer.cdn.mozilla.net/pub/firefox/releases/
+Future<BrowserInstallation> getOrInstallFirefox(
+  String requestedVersion, {
+  StringSink infoLog,
+}) async {
+
+  // Installation support is implemented only for Linux for now.
+  if(!io.Platform.isLinux) {
+    throw UnimplementedError();
```

Comment:
I'll add a comment here.

Reply:
done.

Reply:
Instead of a comment I'd recommend an error message in the exception itself (just copy'n'paste the comment text into the error message) so when it fails you can tell from the logs what's going on.